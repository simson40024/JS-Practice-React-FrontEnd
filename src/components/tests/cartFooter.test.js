import CartFooter from "../cart/cartFooter";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import * as hooks from "../../hook";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const state = {
  currency: {
    currency: "UAH",
    rate: 38,
  },
};

const mock = jest.spyOn(hooks, "useAppSelector");
describe("CartFooter", () => {
  it("should create CartFooter", () => {
    mock.mockImplementation((selector) => selector(state));
    const total = 150;
    act(() => {
      createRoot(container).render(<CartFooter total={total} />);
    });
    expect(container).toMatchSnapshot();
  });
});
