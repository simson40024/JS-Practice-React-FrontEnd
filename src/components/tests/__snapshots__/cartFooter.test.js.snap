// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartFooter should create CartFooter 1`] = `
<div>
  <hr />
  <div
    class="cart-amount"
  >
    <button>
      Confirm order
    </button>
    <p
      class="amount"
    >
      5700
       
      UAH
    </p>
  </div>
</div>
`;
